openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -1,111 +1,161 @@
-        logger.info(f"Results: {str(image_count)}")
-    logger.info(f"Total images pulled: {image_count}")
-    logger.info('Terminated!')
-
-
-if __name__ == '__main__':
-    main()
+SCRIPT_DIR = (
+    "/usr/local/airflow/openverse_catalog/dags/providers/provider_csv_load_scripts/"
+    + "inaturalist/"
+)
+PG_TO_JSON_TEMPLATE = dedent(
+    open(os.path.join(SCRIPT_DIR, "05_export_dbpage_to_json_template.sql")).read()
+)
+
+
+class inaturalistDataIngester(ProviderDataIngester):
+
+    # TO DO: consider using this sql to get a max number of db pages to define a
+    # get_should_continue function
+    # select relpages
+    # from pg_class t join pg_namespace s on t.relnamespace=s.oid
+    # where s.nspname='inaturalist' and t.relname='photos';
+
+    providers = {"image": prov.INATURALIST_DEFAULT_PROVIDER}
+
+    def get_next_query_params(self, old_query_params=None, **kwargs):
+        """Page counter"""
+        if old_query_params is None:
+            return {"page_number": 0}
+        else:
+            next_page = old_query_params["page_number"] + 1
+            return {"page_number": next_page}
+
+    def get_response_json(self, query_params: Dict):
+        """
+        Call the SQL to pull json from Postgres, where the raw data has been loaded.
+        """
+        db_page_number = str(query_params["page_number"])
+        sql_string = PG_TO_JSON_TEMPLATE.replace("db_page_number", db_page_number)
+        pg = PostgresHook(POSTGRES_CONN_ID)
+        return pg.get_records(sql_string)
+
+    def get_batch_data(self, response_json):
+        if response_json:
+            return [dict(item[0]) for item in response_json]
+        return None
+
+    def get_record_data(self, data):
+        license_url = data.get("license_url", None)
+        license_info = get_license_info(license_url=license_url)
+        if license_info == LicenseInfo(None, None, None, None):
+            return None
+
+        foreign_id = data.get("foreign_id")
+        if foreign_id is None:
+            return None
+
+        return {
+            "foreign_identifier": f"{foreign_id}",
+            "foreign_landing_url": data.get("foreign_landing_url"),
+            "title": data.get("title"),
+            "creator": data.get("creator"),
+            "image_url": data.get("image_url"),
+            "width": data.get("width"),
+            "height": data.get("height"),
+            "license_info": license_info,
+            "filetype": data.get("filetype"),
+            "creator_url": data.get("creator_url"),
+            "raw_tags": data.get("tags"),
+        }
+
+    def get_media_type(self, record):
+        # This provider only supports Images via S3, though they have some audio files
+        # on site and in the API.
+        return "image"
+
+    def endpoint(self):
+        assert "Postgres DB" == "API endpoint"
+
+
+def sql_runner(file_name):
+    assert os.path.exists(os.path.join(SCRIPT_DIR, file_name))
+    assert file_name[-4:] == ".sql"
```

Comment:
Similar to the point above, these would be better as raised exceptions:
```suggestion
    if not (SCRIPT_DIR / file_name).exists() or file_name[-4:] != ".sql":
        raise ValueError(f"Provided file is invalid or does not exist: {file_name}")
```

Reply:
Yeah, I think this is taken care of [here](https://github.com/WordPress/openverse-catalog/blob/fe3d0c27142b9d45e304520fc37aa9088f335d99/openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py#L140), right?